src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -348,20 +347,29 @@ private JObject GetVerificationResult(UInt160 scriptHash, ContractParameter[] ar
             {
                 throw new RpcException(-100, "Unknown contract");
             }
-            var methodName = "verify";
+            var md = contract.Manifest.Abi.GetMethod("verify", -1);
+            if (md is null)
+                throw new RpcException(-101, $"The smart contract {contract.Hash} haven't got verify method.");
+            if (md.ReturnType != ContractParameterType.Boolean)
+                throw new RpcException(-102, "The verify method doesn't return boolean value.");
 
-            Transaction tx = signers == null ? null : new Transaction
+            Transaction tx = new Transaction
             {
-                Signers = signers.GetSigners(),
-                Attributes = Array.Empty<TransactionAttribute>()
+                Signers = signers == null ? new Signer[] { new() { Account = scriptHash } } : signers.GetSigners(),
+                Attributes = Array.Empty<TransactionAttribute>(),
+                Witnesses = signers?.Witnesses,
```

Comment:
```suggestion
                Witnesses = signers == null ? new Witness() { InvocationScript = sb.ToArray() } : signers.Witnesses,
```

where `sb` is from
```
if (args.Length > 0)
            {
                using ScriptBuilder sb = new ScriptBuilder();
                for (int i = args.Length - 1; i >= 0; i--)
                    sb.EmitPush(args[i]);
             }
```

Reply:
`sb` will be loaded into engine directly, is this `InvocationScript ` necessary?

Reply:
We know exactly the invocation script, why not to fill it in? By doing this we're getting closer to the real transaction witnesses.